### PR TITLE
Check if Microk8s is installed

### PR DESCRIPTION
**Pull request by eaudetcobello**
Adds a check that uses the snapd API to let the user know microk8s must be uninstalled.

### CODE REVIEW

File: src/k8s/cmd/k8s/k8s_bootstrap.go [hunk 1]
```diff
@@ -47,6 +48,24 @@ func newBootstrapCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
...
+				return
+			}
+			microk8sInfo, err := snapdClient.GetSnapInfo("microk8s")
+			if err != nil {
+				cmd.PrintErrln("Error: failed to check if microk8s is installed: %w", err)
+				env.Exit(1)
+				return
+			}
+			if microk8sInfo.StatusCode == 200 {
+				cmd.PrintErrln("Error: microk8s snap is installed, please remove it and try again.")
```
Comment (HomayoonAlimohammadi) [inline src/k8s/cmd/k8s/k8s_bootstrap.go:64]: Just as a suggestion, maybe we can also guide the user on how to do that? e.g. "... By running `snap remove microk8s`"

File: src/k8s/pkg/client/snapd/snap_info.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+func (c *Client) GetSnapInfo(snap string) (*snapdSnapInfoResponse, error) {
+	resp, err := c.client.Get(fmt.Sprintf("http://localhost/v2/snaps/%s", snap))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get snapd snap info: %w", err)
+	}
+	defer resp.Body.Close()
+
+	resBody, err := io.ReadAll(resp.Body)
```
Comment (HomayoonAlimohammadi) [inline src/k8s/pkg/client/snapd/snap_info.go:20]: Maybe we can used buffered read here?
p.s. The response is probably never that big so feel free to ignore this. Just as a precaution.

**Review by HomayoonAlimohammadi — APPROVED** [review]
Great work! Thanks @eaudetcobello! LGTM, just some minor comments and questions.

**Review by berkayoz — APPROVED** [review]
LGTM, just to make sure we only get response code `200` if snap is installed right? 

We should also mention this in our docs, we can do that in a separate PR/card.

**Comment by HomayoonAlimohammadi** [comment]
Quick question: Do we also care if people have any other thing installed that binds to the ports we care about? E.g. maybe `k3s` or...?

**Comment by eaudetcobello** [comment]
> Quick question: Do we also care if people have any other thing installed that binds to the ports we care about? E.g. maybe `k3s` or...? 

Definitely, we have a card for this.

**Review by addyess — APPROVED** [review]
LGTM.  The queue of suggestions is full but merge when ready

File: src/k8s/pkg/client/snapd/snap_info.go (hunk 2 above)
Comment (eaudetcobello) [reply src/k8s/pkg/client/snapd/snap_info.go:20]: I don't think that's necessary for a few kilobytes (if even that).

File: src/k8s/cmd/k8s/k8s_bootstrap.go (hunk 1 above)
Comment (eaudetcobello) [reply src/k8s/cmd/k8s/k8s_bootstrap.go:64]: Sure

**Pull request merged**